### DISCUSSION TITLE
Having issues setting the role for V8 Buttons

### DISCUSSION THREAD

**Discussion by svalenciasan** [microsoft/fluentui#30518] [Q&A]
Is there any reason why the role might be getting ignored?
```tsx
<>
  <div role='radiogroup'>
    <IconButton
      ariaLabel={strings.AlignAriaLabel}
      data-automation-id={strings.AlignAutomationId}
      iconProps={{ iconName: 'AlignHorizontalLeft' }}
      onClick={() => changeCallback(targetProperty, Alignment.Left)}
      role={'radio'}
      aria-checked={alignment === Alignment.Left}
      styles={defaultButtonStyles}
      title={strings.AlignTitle}
    />
    // ... OTHER ICONBUTTONS WITH SAME ISSUE ...
  </div>
</>

```
![image](https://github.com/microsoft/fluentui/assets/52926410/61e20b55-e8b5-45cd-bd1a-1f6045ec8e6b)

**Comment by pradeept95** [comment]
Hi @svalenciasan ,
When I tried to set role no buttons (DefaultButton, IconButton etc), it seems working properly. Please refer following screenshot from doc.
![image](https://github.com/microsoft/fluentui/assets/28040299/17423bf5-04b7-4733-b219-72e4845dacb2)

` <IconButton role={"radio"} iconProps={emojiIcon} title="Emoji" ariaLabel="Emoji" disabled={disabled} checked={checked} />
`
You can test using the editor in the document itself.
